.clang-tidy

```diff
@@ -0,0 +1,44 @@
+Checks: '*'
+WarningsAsErrors: '*'
+
+HeaderFilterRegex: '^<.*\.hpp>$'
+
+CheckOptions: # TODO(Ri7ay): Setup all options
```

Comment:
@Ri7ay OMG there are insane amount of options, like more than 300 or something around it. I hope that you can handle it.